packages/meditrak-server/src/routes/GETSurveyGroups/assertSurveyGroupsPermissions.js

```diff
@@ -38,16 +41,41 @@ export const filterSurveyGroupsByPermissions = async (accessPolicy, models, surv
   return filteredSurveyGroups;
 };
 
-export const assertSurveyGroupsPermissions = async (accessPolicy, models, surveyGroups) => {
-  const filteredSurveyGroups = await filterSurveyGroupsByPermissions(
-    accessPolicy,
-    models,
-    surveyGroups,
-  );
+export const assertSurveyGroupsPermissions = async (accessPolicy, models, surveyGroupId) => {
+  const surveyGroup = await models.surveyGroup.findById(surveyGroupId);
+  const filteredSurveyGroups = await filterSurveyGroupsByPermissions(accessPolicy, models, [
+    surveyGroup,
+  ]);
 
-  if (filteredSurveyGroups.length !== surveyGroups.length) {
+  if (filteredSurveyGroups.length === 0) {
     throw new Error('You do not have permissions to access the requested survey group(s)');
   }
 
   return true;
 };
+
+export const createSurveyGroupDBFilter = async (accessPolicy, models, criteria) => {
+  if (hasBESAdminAccess(accessPolicy)) {
+    return criteria;
+  }
+  const dbConditions = criteria;
+  const allPermissionGroupsNames = accessPolicy.getPermissionGroups();
+  const countryIdsByPermissionGroupId = {};
+
+  // Generate lists of country ids we have access to per permission group id
+  for (const pg of allPermissionGroupsNames) {
+    const permissionGroup = await models.permissionGroup.findOne({ name: pg });
+    if (permissionGroup) {
+      const countryNames = accessPolicy.getEntitiesAllowed(pg);
+      const countryList = await models.country.find({ code: countryNames });
+      countryIdsByPermissionGroupId[permissionGroup.id] = countryList.map(e => e.id);
+    }
+  }
+
+  // Check we have permissions to at least one of the surveys in this group
+  dbConditions[RAW] = {
+    sql: `(select count(*) from survey where survey.survey_group_id = survey_group.id and country_ids && array(select trim('"' from json_array_elements(?::json->permission_group_id)::text))) > 0`,
```

Comment:
Again just padding it out with indentation and new lines

Have you done any performance comparisons against dev at all? I wonder whether adding a select for every row adds much of a hit.